guidance/library/_json.py

```diff
@@ -718,10 +703,182 @@ def anyOf(
         lm,
         *,
         anyof_list: Sequence[JSONSchema],
+        base_uri: str,
     ):
-        options = [self.json(json_schema=item) for item in anyof_list]
+        options = [self.json(json_schema=item, base_uri=base_uri) for item in anyof_list]
         return lm + select(options)
 
+    @guidance(stateless=True)
+    def oneOf(
+        self,
+        lm,
+        *,
+        oneof_list: Sequence[JSONSchema],
+        base_uri: str,
+    ):
+        if len(oneof_list) == 1:
+            return lm + self.json(json_schema=oneof_list[0], base_uri=base_uri)
+        warnings.warn("oneOf not fully supported, falling back to anyOf. This may cause validation errors in some cases.")
+        return lm + self.anyOf(anyof_list=oneof_list, base_uri=base_uri)
+
+    @guidance(stateless=True)
+    def allOf(
+        self,
+        lm,
+        *,
+        parent_schema: JSONSchema,
+        base_uri: str,
+    ):
+        type = set(JSONType)
+        properties: defaultdict[str, list[JSONSchema]] = defaultdict(list)
+        required: set[str] = set()
+        additional_properties_list: list[JSONSchema] = []
+        items_list: list[JSONSchema] = []
+        other_data: dict[str, Any] = {}
+        enum: Optional[list[Any]] = None
+        const: Union[Unset, Any] = _unset
+
+        def handle_keyword(key: str, value: Any, base_uri: str):
+            nonlocal type
+            nonlocal required
+            nonlocal const
+            nonlocal enum
+
+            if key == Keyword.REF:
+                ref = cast(str, value)
+                abspath = urijoin(base_uri, ref)
+                resolved = self._resolver.lookup(abspath)
+                add_schema(resolved.contents, base_uri=resolved.resolver._base_uri)
+
+            elif key == Keyword.CONST:
+                if const is not _unset and const != value:
+                    raise ValueError(f"allOf with multiple conflicting const values: {const!r} and {value!r}")
+                const = value
+
+            elif key == Keyword.ENUM:
+                value = cast(list[Any], value)
+                if enum is not None:
+                    try:
+                        enum = list(set(enum) & set(value))
+                    except TypeError:
+                        # Check on equality, not on hash
+                        # Yes, this is O(n^2).
+                        # Hope the items were unique.
```

Comment:
This line makes me wonder if we should just raise our own exception (with the type error as an inner_exception), explaining what's happened and showing the two offending lists?

Reply:
Annoyingly, you can have an enum over objects (read: dicts). Set intersection will only work if the objects are hashable. Otherwise we're falling back to intersecting enums based on equality, which shouldn't be too awful for short lists. I'd rather not throw an exception here since we can technically handle it, just suboptimally...

Reply:
Oh just noticed too -- I need to add an exception for the case where we end up with an empty enum...

Reply:
OK